### PR TITLE
Allow gridpack execution to override runtime environment also for legacy gridpacks

### PR DESCRIPTION
**Pull request by bendavid**
This is needed in order to allow reuse of existing gridpacks from PhaseIISummer17wmLHEGENOnly campaign (CMSSW_9_1_2_patch1 slc6_amd64_gcc530) in HGCal tdr campaign (PhaseIITDRFall17wmLHEGS slc6_amd64_gcc630) since remaking them would be a large effort and there are no other changes to the contents needed.

### CODE REVIEW

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
please test

**Comment by perrozzi** [comment]
+1

File: GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh [hunk 1]
```diff
@@ -26,10 +26,36 @@ echo "%MSG-MG5 random seed used for the run = $rnum"
...
+  cmssw_version=${7}
+  echo "%MSG-MG5 override cmssw_version = $cmssw_version"
+fi
 
 LHEWORKDIR=`pwd`
 
+if [ "$use_gridpack_env" = false -a -n "$scram_arch_version" -a -n  "$cmssw_version" ]; then
+  echo "%MSG-MG5 CMSSW version = $cmssw_version"
+  export VO_CMS_SW_DIR=/cvmfs/cms.cern.ch
+  source $VO_CMS_SW_DIR/cmsset_default.sh
```
Comment (davidlange6) [inline GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh]: hi @bendavid - i'm surprised that these two lines would be needed? the job is running within a cms environment after all

Comment (bendavid) [reply GeneratorInterface/LHEInterface/data/run_generic_tarball_cvmfs.sh]: This is handling the case where a site has configured a non-cvmfs source for the release area (where there would presumably be weaker guarantees on which releases are available)

**Comment by davidlange6** [comment]
i think the software needs to let the site config determine which CMSSW software installation should be used at runtime. (yes, obviously none of this will be used at a non-cvmfs site in the short term)

**Comment by bendavid** [comment]
Ok, newer gridpacks are already doing that themselves (e.g. https://github.com/cms-sw/genproductions/blob/master/bin/MadGraph5_aMCatNLO/runcmsgrid_LO.sh#L36-L42)  but I can remove those lines from here in any case.

**Comment by davidlange6** [comment]
please test


On 28 Sep 2017, at 10:31, cmsbuild <notifications@github.com<mailto:notifications@github.com>> wrote:


Pull request #20676<https://github.com/cms-sw/cmssw/pull/20676> was updated. @cmsbuild<https://github.com/cmsbuild>, @efeyazgan<https://github.com/efeyazgan>, @perrozzi<https://github.com/perrozzi>, @thuer<https://github.com/thuer>, @govoni<https://github.com/govoni> can you please check and sign again.

**Comment by perrozzi** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**